forgerock-auth/src/main/java/org/forgerock/android/auth/webauthn/WebAuthKeySelectionDialogFragment.java

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright (c) 2021 ForgeRock. All rights reserved.
+ *
+ * This software may be modified and distributed under the terms
+ * of the MIT license. See the LICENSE file for details.
+ */
+
+package org.forgerock.android.auth.webauthn;
+
+import android.content.DialogInterface;
+import android.os.Bundle;
+import android.os.OperationCanceledException;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.AdapterView;
+import android.widget.ArrayAdapter;
+import android.widget.Button;
+import android.widget.Spinner;
+import android.widget.TextView;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.fragment.app.DialogFragment;
+
+import org.forgerock.android.auth.FRListener;
+import org.forgerock.android.auth.Listener;
+import org.forgerock.android.auth.R;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * A Simple Dialog to display key selection using a spinner
+ */
+public class WebAuthKeySelectionDialogFragment extends DialogFragment {
+
+    private static final String PUBLIC_KEY_CREDENTIAL_SOURCE = "PUBLIC_KEY_CREDENTIAL_SOURCE";
+    private boolean isCancel = true;
+    private List<PublicKeyCredentialSource> sources;
+    private FRListener<PublicKeyCredentialSource> listener;
+
+    public WebAuthKeySelectionDialogFragment() {
```

Comment:
Should we provide a javadoc for this constructor?

Reply:
Usually I don't, but no harm